src/test/java/io/cdap/plugin/http/source/common/pagination/PaginationIteratorTest.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.http.source.common.pagination;
+
+import io.cdap.plugin.http.source.batch.HttpBatchSourceConfig;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PaginationIteratorTest {
```

Comment:
For now I have removed this test, since deadlines are extra tight and this will take some time, to mock all the pagination types and implement the mock server and also all the data types. I will maybe implement it later. Also I will ask Albert which kind of tests do they prefer, so I have a knowledge what to implement if they have different vision of this.